packages/platform-browser/src/dom/shared_styles_host.ts

```diff
@@ -6,12 +6,28 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {DOCUMENT} from '@angular/common';
-import {APP_ID, Inject, Injectable, OnDestroy} from '@angular/core';
+import {DOCUMENT, isPlatformServer} from '@angular/common';
+import {APP_ID, Inject, Injectable, OnDestroy, PLATFORM_ID} from '@angular/core';
 
 @Injectable()
 export class SharedStylesHost implements OnDestroy {
-  private readonly usageCount = new Map<string /** Style string */, number /** Usage count */>();
+  // Maps all registered host nodes to a list of style nodes that have been added to the host node.
+  private readonly styleRef = new Map < string /** Style string */, {
+    elements: HTMLStyleElement[];
+    usage: number
+  }
+  > ();
+  private hostNodes = new Set<Node>();
+  private styleNodesInDOM: Map<string, HTMLStyleElement>|null;
+  private platformIsServer: boolean;
+
+  constructor(
+      @Inject(DOCUMENT) private readonly doc: Document, @Inject(APP_ID) private appId: string,
+      @Inject(PLATFORM_ID) platformId: object = {}) {
```

Comment:
We can use this change as an opportunity to switch to the `inject` function, WDYT?
```suggestion
  private readonly doc = inject(DOCUMENT);
  private readonly appId = inject(APP_ID);
  platformId = inject(PLATFORM_ID); // should it also be private and readonly?
  private readonly isPlatformServer = isPlatformServer(this.platformId);

  constructor() {
```

Reply:
I was going to do that, but in some cases, this classes is initialised using the ctor ex: https://github.com/angular/angular/pull/49424/files#diff-bd84358b34fb4defc3222724ec52e393a302cb7d3a3df8282415ae8f53a3c7d6R43, so it would to be changed to 

```ts
  doc: Document;

  constructor(
      doc?: Document,
      appId?: string,
      platformId?: object
    ) {

    this.doc = doc ?? inject(DOCUMENT);
    ...
  }
```

Do you still prefer `inject`?

Reply:
Thanks for the comment. No worries, let's keep things as is then. The mentioned usage comes from the tests and would require some additional refactoring (create an injector and run a function within its context to get a hold of the `SharedStylesHost` token), which we can do afterwards (anytime, after v16).